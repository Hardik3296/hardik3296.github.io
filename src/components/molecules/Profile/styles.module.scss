@import "../../../fonts.scss";

.row{
   display:flex;
   flex-direction: row;
   flex:1;
   height: 100vh;
   @media (max-width:1024px){
      padding-top:5vh;
      flex-direction: column;
      align-items: center;
      justify-content: center;
    }
   
   align-items: center;
   .firstDiv{
      display:flex;
      flex:2;
      flex-direction: column;
      justify-content: center;
      align-items: center;
      @media (max-width:1024px) {
         flex:1
      }
      .image{
         height: 20vw;
         width: 20vw;
         border-radius: 20vw;
         @media (max-width:1024px){
            height: 40vw;
            width: 40vw;
            border-radius: 40vw;
         }
      }
   }
   .secondDiv{
      .name{
         font-size:75px;
         margin-bottom:20px;
         @media (max-width:1024px){
            font-size: 13vw;
            margin-bottom:5vw;
         }
         font-family: ArchivoMedium;
         width: fit-content;
      }
      flex:3;
      @media (max-width:1024px){
         flex:1;
         display:flex;
         flex-direction:column;
         justify-content: flex-start;
         align-items: center;
      }
      .contact{
         display:flex;
         align-items:center;
         width: fit-content;
         @media(max-width:1024px){
            width:100%;
            justify-content:space-around;
         }
         a:first-child{
         margin-left:0px;
         }
         a{
            margin-left:50px;
         }
      }
   }
}