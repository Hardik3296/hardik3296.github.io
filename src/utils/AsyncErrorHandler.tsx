import React from "react";

const useAsyncError = () => {
   const [_error, setError] = React.useState();
   return React.useCallback(
      e => {
         setError(() => {
            throw e;
         });
      },
      [setError],
   );
};

export default useAsyncError;