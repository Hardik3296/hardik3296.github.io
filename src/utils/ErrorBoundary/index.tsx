import { Component, ReactNode } from "react";
import styles from "./styles.module.scss";

interface MyProps {
   textcolor: string,
   children: ReactNode,
}

class ErrorBoundary extends Component<MyProps, {}> {

   state = {
      hasError: false
   }

   static getDerivedStateFromError() {
      return { hasError: true }
   }

   componentDidCatch(error: any, errorInfo: any) {
   }
   render() {
      if (this.state.hasError)
         return <div className={styles.errorDiv}>
            <p style={{ color: this.props.textcolor }}>Something went wrong while fetching the repositories!!</p>
         </div>
      else
         return this.props.children;
   }
}

export default ErrorBoundary;