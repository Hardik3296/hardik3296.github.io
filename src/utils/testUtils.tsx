import { ShallowWrapper } from "enzyme"

export const findAttributeByTestId = (component:ShallowWrapper, attr:string):ShallowWrapper=>{
   return component.find(`[test-id='${attr}']`);
}