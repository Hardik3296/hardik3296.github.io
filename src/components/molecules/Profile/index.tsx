import { Dispatch, memo, MutableRefObject, SetStateAction, useContext } from "react";
import image from "../../../assets/images/me.jpg";
import styles from "./styles.module.scss";
import SocialMedia from "../Social-Media";
import { ThemeContext, ThemeInterface } from "../../../utils/contexts/ThemeContext";
import SlideAnimation from "../../atoms/SlideAnimation";
import ErrorBoundary from "../../../utils/ErrorBoundary";

const Profile = ({ profileRef }: { profileRef: MutableRefObject<HTMLDivElement> }): JSX.Element => {

   const [theme, _setTheme] = useContext<[ThemeInterface, Dispatch<SetStateAction<ThemeInterface>>]>(ThemeContext);

   return (
      <div className={styles.row} id="me" style={theme.text} ref={profileRef}>
         <div className={styles.firstDiv} test-id="image-div">
            <ErrorBoundary textcolor={theme.error.text.color}>
               <SlideAnimation slideDirection={"slide-in-from-left"}>
                  <img src={image} alt="me" className={styles.image} />
               </SlideAnimation>
            </ErrorBoundary>
         </div>
         <div className={styles.secondDiv}>
            <ErrorBoundary textcolor={theme.error.text.color}>
               <SlideAnimation slideDirection={"slide-in-from-right"}>
                  <p className={styles.name} test-id="name">Hardik Gaur</p>
               </SlideAnimation>
            </ErrorBoundary>
            <ErrorBoundary textcolor={theme.error.text.color}>
               <SlideAnimation
                  slideDirection={"slide-in-from-bottom"}
               >
                  <div className={styles.contact} test-id="social-icons">
                     <SocialMedia size={50} icons={theme.icons} />
                  </div>
               </SlideAnimation>
            </ErrorBoundary>
         </div>
      </div>
   );
}

export default memo(Profile);