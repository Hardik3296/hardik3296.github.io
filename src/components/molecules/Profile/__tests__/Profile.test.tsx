import { MutableRefObject, createRef } from "react";
import { shallow, ShallowWrapper } from "enzyme";
import Profile from "..";
import renderer from "react-test-renderer";
import { findAttributeByTestId } from "../../../../utils/testUtils";

const ref = createRef<MutableRefObject<HTMLDivElement>>();

const renderComponent = (): ShallowWrapper => {
   return shallow(<Profile profileRef={ref} />);
}

let wrapper: ShallowWrapper;

describe("Profile component", () => {

   beforeEach(() => {
      wrapper = renderComponent();
   });

   it("Should have 1 image div", () => {
      const component = findAttributeByTestId(wrapper, "image-div");
      expect(component.length).toBe(1);
   });

   it("Should have 1 name ", () => {
      const component = findAttributeByTestId(wrapper, "name");
      expect(component.length).toBe(1);
   });

   it("Should have 1 social icons", () => {
      const component = findAttributeByTestId(wrapper, "social-icons");
      expect(component.length).toBe(1);
   });

   it("Should render correcty", () => {
      const component = renderer.create(<Profile profileRef={ref} />).toJSON();
      expect(component).toMatchSnapshot();
   })
});