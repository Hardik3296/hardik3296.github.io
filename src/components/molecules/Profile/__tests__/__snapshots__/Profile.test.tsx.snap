// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Profile component Should render correcty 1`] = `
<div
  className="row"
  id="me"
  style={
    Object {
      "color": "#000000",
    }
  }
>
  <div
    className="firstDiv"
    test-id="image-div"
  >
    <div
      className="errorDiv"
    >
      <p
        style={
          Object {
            "color": "#B00020",
          }
        }
      >
        Something went wrong while fetching the repositories!!
      </p>
    </div>
  </div>
  <div
    className="secondDiv"
  >
    <div
      className="errorDiv"
    >
      <p
        style={
          Object {
            "color": "#B00020",
          }
        }
      >
        Something went wrong while fetching the repositories!!
      </p>
    </div>
    <div
      className="errorDiv"
    >
      <p
        style={
          Object {
            "color": "#B00020",
          }
        }
      >
        Something went wrong while fetching the repositories!!
      </p>
    </div>
  </div>
</div>
`;
